ocaml/xenopsd/lib/xenops_server.ml

```diff
@@ -2628,19 +2629,23 @@ and perform_exn ?result (op : operation) (t : Xenops_task.task_handle) : unit =
           ~path:(Uri.path_unencoded url ^ snippet ^ id_str)
           ~query:(Uri.query url) ()
       in
-      (* CA-78365: set the memory dynamic range to a single value to stop
-         ballooning. *)
-      let atomic =
-        VM_set_memory_dynamic_range
-          (id, vm.Vm.memory_dynamic_min, vm.Vm.memory_dynamic_min)
-      in
-      let (_ : unit) =
-        perform_atomic ~progress_callback:(fun _ -> ()) atomic t
-      in
-      (* Waiting here is not essential but adds a degree of safety and
-         reducess unnecessary memory copying. *)
-      ( try B.VM.wait_ballooning t vm
-        with Xenopsd_error Ballooning_timeout_before_migration -> ()
+      (* CA-78365: set the memory dynamic range to a single value to stop ballooning.
+
+         There's no need to balloon down when doing localhost migration - we're
+         not copying any memory in the first place. This would likely increase VDI
```

Comment:
You may not have enough host memory to complete the migration though, although I agree that ballooning in general is not a good idea.

Although on a (network) migration it'd reduce the amount of data you need to send and cut down on migration time, it'll likely cause the domain on the other side to then be created with Populate on Demand, which is known to be unreliable.

I'd say that usually localhost migration is not something one would expect to optimize for, but then I see you mentioning VDI, and I remembered that we still do a full live migration of the VM even if we only need to migrate its VDI. There is a comment in the code somewhere that we should instead implement fast-resume (then I think we could avoid the double-the-memory requirement too).